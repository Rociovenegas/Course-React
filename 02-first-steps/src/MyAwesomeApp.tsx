import React from 'react';


type MyAwesomeAppProps = {
    name: string;
    lastName: string;
};

const myStyles: React.CSSProperties = {
    backgroundColor: '#6a6afa',
    borderRadius: 20,
    padding: 10,
    marginTop: 30,
};

const address = {
    zipCode: 'ABC-123',
    country: 'CanadÃ¡',
};

export const MyAwesomeApp = (props: MyAwesomeAppProps) => {
    return (
        <div data-testid='div-app-01'>
            <h1 data-testid='Test-01'> Mongi </h1>
            <h3 className='mi-clase'> Hermosa </h3>
            <h1>Hello, {props.name}!</h1>
            <h3>{props.lastName} :3</h3>



            <p style={myStyles}>{JSON.stringify(address)}</p>
        </div>
    );
};