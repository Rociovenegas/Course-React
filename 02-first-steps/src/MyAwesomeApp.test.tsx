import { describe, expect, test } from "vitest";
import { render, screen } from '@testing-library/react';
import { MyAwesomeApp } from "./MyAwesomeApp";

describe('MyAwesomeApp', () => {
    test('should render name', () => {
        const { container } = render(<MyAwesomeApp name='Gaby' lastName='Jhonny' />);
        //console.log(container.innerHTML)

        const h1 = container.querySelector('h1');
        expect(h1?.innerHTML).toContain('Mongi')
        const h3 = container.querySelector('h3');
        expect(h3?.innerHTML).toContain('Hermosa')
    })

    test('should render name - screen', () => {
        render(<MyAwesomeApp name='Gaby' lastName='Jhonny' />);
        screen.debug()

        const h1 = screen.getByTestId('Test-01');
        expect(h1.innerHTML).toContain('Mongi');

    })

    test('should match snapshot', () => {
        const { container } = render(<MyAwesomeApp name='Gaby' lastName='Jhonny' />);
        expect(container).toMatchSnapshot();
    })

    test('should match snapshot - screen', () => {
        render(<MyAwesomeApp name='Gaby' lastName='Jhonny' />);
        expect(screen.getByTestId('div-app-01')).toMatchSnapshot();
    })
});

//div-app-01