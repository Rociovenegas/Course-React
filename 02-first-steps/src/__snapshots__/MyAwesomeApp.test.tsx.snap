// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MyAwesomeApp > should match snapshot - screen 1`] = `
<div
  data-testid="div-app-01"
>
  <h1
    data-testid="Test-01"
  >
     Mongi 
  </h1>
  <h3
    class="mi-clase"
  >
     Hermosa 
  </h3>
  <h1>
    Hello, 
    Gaby
    !
  </h1>
  <h3>
    Jhonny
     :3
  </h3>
  <p
    style="background-color: rgb(106, 106, 250); border-radius: 20px; padding: 10px; margin-top: 30px;"
  >
    {"zipCode":"ABC-123","country":"Canadá"}
  </p>
</div>
`;

exports[`MyAwesomeApp > should match snapshot 1`] = `
<div>
  <div
    data-testid="div-app-01"
  >
    <h1
      data-testid="Test-01"
    >
       Mongi 
    </h1>
    <h3
      class="mi-clase"
    >
       Hermosa 
    </h3>
    <h1>
      Hello, 
      Gaby
      !
    </h1>
    <h3>
      Jhonny
       :3
    </h3>
    <p
      style="background-color: rgb(106, 106, 250); border-radius: 20px; padding: 10px; margin-top: 30px;"
    >
      {"zipCode":"ABC-123","country":"Canadá"}
    </p>
  </div>
</div>
`;
